### ISSUE TITLE
The mark is shifted when a new heading is added

### ISSUE THREAD

**Issue by Andreas-AL** [splitbrain/dokuwiki#2247]
Method:
  - Create new page (preferably a blank page)
  - Move the cursor to the last character.
  - Select the button with the same level headline [8].
  - Move the cursor to the **last** character.
  - Enter
- Select the button "Upper heading [0].

  After this procedure the problem always appears that the marked text is shifted by two places ...

![test](https://user-images.githubusercontent.com/36191954/35856115-e1fb7d88-0b2c-11e8-9e2d-5fe7bd6c6de6.jpg)

**Comment by splitbrain** [comment]
Thanks for opening the issue. Unfortunately your issue does not contain enough information. Please try to provide as much info as possible, generally following these three main questions:

* What did you do?
* What did you expect to happen?
* What happened instead?

Please also describe what steps you took to analyze the issue and what (failed) attempts you made to fix the problem yourself.

Your issue should contain everything that might help in analyzing and reproducing the issue. That includes but isn't limited to:

* versions of the involved software
* exact error messages
* debug logs (also check your JavaScript console for web related issues)

Thanks for your help.

PS: You may want to refer to [How to Ask Questions the Smart Way](http://www.catb.org/esr/faqs/smart-questions.html) for more tips on improving your issue creation game ;-)

**Comment by phy25** [comment]
I can reproduce this in IE11 on dokuwiki.org. Indeed it is hard to reproduce this if I change a little bit among the steps.

![Shifted selection area](https://user-images.githubusercontent.com/331403/35901689-e5d9548e-0c12-11e8-815d-31e2b487a908.png)

**Comment by Andreas-AL** [comment]
Yes that's it. You have to try out a little until you find the bug.
I have not yet found out in which regularity the error occurs.

**Comment by Klap-in** [comment]
Some general pointers to the code of this header buttons. Maybe useful for you, now you could reproduce the behavior in certain manner..

https://github.com/splitbrain/dokuwiki/blob/master/lib/scripts/toolbar.js
https://github.com/splitbrain/dokuwiki/blob/master/lib/scripts/textselection.js

definition of toolbar buttons, outputted as json: https://github.com/splitbrain/dokuwiki/blob/master/inc/toolbar.php
when called in: https://github.com/splitbrain/dokuwiki/blob/master/lib/exe/js.php
